client/src/common/client.ts

```diff
@@ -922,13 +922,26 @@ export abstract class BaseLanguageClient implements FeatureClient<Middleware, La
 	}
 
 	public async start(): Promise<void> {
+		return this.startInternal(true);
+	}
+
+	private async startInternal(allowRestart = false): Promise<void> {
 		if (this._onStart !== undefined) {
 			return this._onStart;
 		}
 		if (this._onStop !== undefined) {
 			await this._onStop;
 			this._onStop = undefined;
 		}
+
+		if (!allowRestart) {
+			this.outputChannel.appendLine(`***** already stopped. client ${this.name}`);
+
+			throw new LSPCancellationError({ retriggerRequest: false });
```

Comment:
set breakpoint here.

I made this change so that client fails rather than silently restart server that is already explicitly stopped.